### ISSUE TITLE
panic:cmd failed: 252: /bin/zsh -i --no-rcs --no-globalrcs

### ISSUE THREAD

**Issue by bortels** [teaxyz/cli#189]
It says I have to report the bug, so - ok, sure. Fair's fair. 

```
Error: cmd failed: 252: /bin/zsh -i --no-rcs --no-globalrcs
    at run (file:///opt/tea.xyz/src/v0.13.8/src/utils/index.ts:186:34)
    at async repl (file:///opt/tea.xyz/src/v0.13.8/src/app.exec.ts:284:5)
    at async default (file:///opt/tea.xyz/src/v0.13.8/src/app.exec.ts:21:9)
    at async file:///opt/tea.xyz/src/v0.13.8/src/app.ts:31:9
```

And - it's reproducible, which is fun. Here's what I did (I have the "aws" command installed externally, probably from AWS's pkg installer:

```
tbortels@Toms-Mac-mini ~ % tea +python.org
this is a temporary shell containing the following packages:
invisible-island.net/ncurses@6.3.0, gnu.org/readline@8.1.0, zlib.net@1.2.13, sourceware.org/bzip2@1.0.8, openssl.org@1.1.118, sourceware.org/libffi@3.4.3, libexpat.github.io@2.5.0, bytereef.org/mpdecimal@2.5.1, tukaani.org/xz@5.2.7, sqlite.org@3.39.4, python.org@3.11.0, pip.pypa.io@22.3.0
when done type: `exit'
tea ~ aws

usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help

aws: error: the following arguments are required: command

tea ~ exit

panic: split tea. we’re sorry and we’ll fix it… but you have to report the bug!

    https://github.com/teaxyz/cli/issues/new?title=panic%3Acmd+failed%3A+252%3A+%2Fbin%2Fzsh+-i+--no-rcs+--no-globalrcs

----------------------------------------------------->> attachment begin
Error: cmd failed: 252: /bin/zsh -i --no-rcs --no-globalrcs
    at run (file:///opt/tea.xyz/src/v0.13.8/src/utils/index.ts:186:34)
    at async repl (file:///opt/tea.xyz/src/v0.13.8/src/app.exec.ts:284:5)
    at async default (file:///opt/tea.xyz/src/v0.13.8/src/app.exec.ts:21:9)
    at async file:///opt/tea.xyz/src/v0.13.8/src/app.ts:31:9
<<------------------------------------------------------- attachment end

```

If I don't run "aws" - it exits quietly.

**Comment by AkiraHasegawa1997** [comment]
Interestingly, the same thing is occurred by executing an invalid random command like the following:
```terminal
➜ tea +python.org
this is a temporary shell containing the following packages:
invisible-island.net/ncurses@6.3.0, gnu.org/readline@8.1.0, zlib.net@1.2.13, sourceware.org/bzip2@1.0.8, openssl.org@1.1.118, sourceware.org/libffi@3.4.3, libexpat.github.io@2.5.0, bytereef.org/mpdecimal@2.5.1, tukaani.org/xz@5.2.7, sqlite.org@3.39.4, python.org@3.11.0, pip.pypa.io@22.3.0
when done type: `exit'
tea ~ lorem ipsum
zsh: command not found: lorem
tea ~ exit

panic: split tea. we’re sorry and we’ll fix it… but you have to report the bug!

    https://github.com/teaxyz/cli/issues/new?title=panic%3Acmd+failed%3A+127%3A+%2Fbin%2Fzsh+-i+--no-rcs+--no-globalrcs

----------------------------------------------------->> attachment begin
Error: cmd failed: 127: /bin/zsh -i --no-rcs --no-globalrcs
    at run (file:///opt/tea.xyz/src/v0.13.6/src/utils/index.ts:183:34)
    at async repl (file:///opt/tea.xyz/src/v0.13.6/src/app.exec.ts:286:5)
    at async default (file:///opt/tea.xyz/src/v0.13.6/src/app.exec.ts:21:9)
    at async file:///opt/tea.xyz/src/v0.13.6/src/app.ts:31:9
<<------------------------------------------------------- attachment end
```

I guess when commands are failed to run, the error will be reported.

**Comment by bortels** [comment]
good observation. 

And going in, running an error command ("lorem"), then a good command ("ls") clears it, so it looks like maybe the last command status is propagating back out? Maybe part of the pipelining?

**Comment by AkiraHasegawa1997** [comment]
I agree.

If running with verbose option and intentionally creating exit status, it's reflected to the status and code
```terminal
➜ tea -vvv +python.org
{
  args: { args: [], pkgs: [ { project: "python.org", constraint: * } ] },
  flags: {
    verbosity: 3,
    magic: true,
    json: false,
    numpty: false,
    sync: false,
    verbose: true,
    debug: true,
    muggle: false,
    silent: false
  }
}
...
tea ~/testdir bash -c "exit 101"
tea ~/testdir exit
{ exit: { success: false, code: 101 } }

panic: split tea. we’re sorry and we’ll fix it… but you have to report the bug!

    https://github.com/teaxyz/cli/issues/new?title=panic%3Acmd+failed%3A+101%3A+%2Fbin%2Fzsh+-i+--no-rcs+--no-globalrcs

----------------------------------------------------->> attachment begin
Error: cmd failed: 101: /bin/zsh -i --no-rcs --no-globalrcs
    at run (file:///opt/tea.xyz/src/v0.13.6/src/utils/index.ts:183:34)
    at async repl (file:///opt/tea.xyz/src/v0.13.6/src/app.exec.ts:286:5)
    at async default (file:///opt/tea.xyz/src/v0.13.6/src/app.exec.ts:21:9)
    at async file:///opt/tea.xyz/src/v0.13.6/src/app.ts:31:9
------------------------------------------------------------------------
{
  err: Error: cmd failed: 101: /bin/zsh -i --no-rcs --no-globalrcs
    at run (file:///opt/tea.xyz/src/v0.13.6/src/utils/index.ts:183:34)
    at async repl (file:///opt/tea.xyz/src/v0.13.6/src/app.exec.ts:286:5)
    at async default (file:///opt/tea.xyz/src/v0.13.6/src/app.exec.ts:21:9)
    at async file:///opt/tea.xyz/src/v0.13.6/src/app.ts:31:9
}
<<------------------------------------------------------- attachment end
```

**Comment by jhheider** [comment]
Yeah, this seems confusing. Most likely the repl should swallow individual exit codes and only abend if tea has an issue.

**Comment by mxcl** [comment]
yeah what is the “generally considered correct behavior here”.

1. obv. we don’t show the error message
2. but should we propagate the error code out of the “repl”?

I'm thinking both, and will PR that but thoughts welcome.

**Comment by mxcl** [comment]
k fix coming up in 0.13.9
